### PR TITLE
move project to rust

### PR DESCRIPTION
**Pull request by barnjamin**

### CODE REVIEW

File: python/util.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+def mul(lhs: int, rhs: int) -> int:
+    o64 = wrapped_mul(lhs, rhs, 64)
+    cast_down_o64 = int.from_bytes(o64.to_bytes(8, "big")[4:], "big")
+    low = wrapped_sub(0, cast_down_o64, 32)
+    red = wrapped_mul(M, low, 32)
+    o64 += wrapped_mul(red, PRIME, 64)
+    ret = o64 >> 32
+    if ret > PRIME:
+        return ret - PRIME
```
Comment (tzaffi) [inline python/util.py:100]:
```suggestion
        return ret % PRIME
```
Probly my suggestion is wrong, and I don't see the big picture, but the most obvious thing to do if you're trying to get the product in the range $[0, \rm{PRIME})$ is to **mod**

File: python/util.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+
+def add(lhs: int, rhs: int) -> int:
+    x = wrapped_add(lhs, rhs, 32)
+    return x if x < PRIME else wrapped_sub(x, PRIME, 32)
+
+
+def sub(lhs: int, rhs: int) -> int:
+    x = wrapped_sub(lhs, rhs, 32)
+    return x if x >= PRIME else wrapped_add(x, PRIME, 32)
```
Comment (ahangsu) [inline python/util.py:89]: this line looks off, maybe it should be 
```suggestion
    return x if x < PRIME else wrapped_add(x, PRIME, 32)
```

File: python/util.py (hunk 1 above)
Comment (barnjamin) [reply python/util.py:100]: The suggestion seems correct, the above I think was copied directly from risc0 core but maybe they do the compare first to save ops? I'll commit this locally and test that I get the same answers

File: python/util.py (hunk 2 above)
Comment (barnjamin) [reply python/util.py:89]: The suggestion looks right, I've no idea how I happened to get the right answers 🤔 will test locally

**Pull request merged**